docs/en/stores/basic-usage.md

```diff
@@ -0,0 +1,471 @@
+-   **Processes** execute a group of commands and represent application behavior
+
+### Operations
+
+Stores cannot be modified directly. Instead four operations are available for modifying state: `add`, `remove`, `replace`, and `test`. These operations are represented by simple objects based on the [JSON patch](http://jsonpatch.com/) format, but in practice helper functions will always be used.
+
+#### `path` and `at`
+
+The path is a `string` that describes the location where an operation is applied. The `path` and `at` functions are part of the `Store` and are accessible inside of a `Command` (which we will talk about later).
+
+Here are a couple examples to show how `path` and `at` describe a location in the store. The `State` is the same as defined above in `interface.d.ts`. It is used by the `Store` to understand the shape of the state data.
+
+> `path` of the current user name
+
+```ts
+const store = new Store<State>();
+const { path } = store;
+
+path('users', 'current', 'name');
+```
+
+This path refers to the `string` value located at `/users/current/name`. `path` is used to transverse the hierarchy in a type-safe way. It ensures only the property names defined in the `State` interface are used.
+
+> `at` is an offset
+
+```ts
+const store = new Store<State>();
+const { at, path } = store;
+
+at(path('users', 'list'), 1);
+```
+
+This path refers to the `User` located at `/users/list` at offset `1`.
+
+#### `add` operation
+
+Adds a value to an object or inserts it into an array.
+
+> `add` example
+
+```ts
+import { add } from '@dojo/framework/stores/state/operations';
+
+const store = new Store<State>();
+const { apply, at, path } = store;
+const user = { id: '0', name: 'Paul' };
+
+apply([add(at(path('users', 'list'), 0), user)]);
+```
+
+This will add `user` to the beginning of the user list.
+
+#### `remove` operation
+
+Removes a value from an object or an array
+
+> `remove` example
+
+```ts
+import { add, remove } from '@dojo/framework/stores/state/operations';
+
+const store = new Store<State>();
+const { apply, at, path } = store;
+const user = { id: '0', name: 'Paul' };
+
+apply([
+	add(path('users'), {
+		current: user,
+		list: [user]
+	}),
+	remove(at(path('users', 'list'), 0))
+]);
+```
+
+This example will add an initial state for `users` and remove the first `user` in the list.
+
+#### `replace` operation
+
+Replaces a value. Equivalent to a `remove` followed by an `add`.
+
+> `replace` example
+
+```ts
+import { add, replace } from '@dojo/framework/stores/state/operations';
+
+const store = new Store<State>();
+const { apply, at, path } = store;
+const users = [{ id: '0', name: 'Paul' }, { id: '1', name: 'Michael' }];
+const newUser = { id: '2', name: 'Shannon' };
+
+apply([
+	add(path('users'), {
+		current: user[0],
+		list: users
+	}),
+	replace(at(path('users', 'list'), 1), newUser)
+]);
+```
+
+This example will replace the second user in the `list` with `newUser`.
+
+#### `test` operation
```

Comment:
I don't think the `test` operation really belongs in basic usage, generally users won't need to explicitly use the `test` operation.